src/Analyser/NodeScopeResolver.php

```diff
@@ -3310,6 +3310,9 @@ private function processAssignVar(
 				if ($propertyReflection->canChangeTypeAfterAssignment()) {
 					$scope = $scope->assignExpression($var, $assignedExprType);
 				}
+				if ($assignedExprType->isSuperTypeOf($propertyReflection->getWritableType())->no()) {
+					$throwPoints[] = ThrowPoint::createImplicit($scope, $assignedExpr);
```

Comment:
The error might occur if we have a property with a native type `int` and we try to assign `int|string` into it. It means that the condition should be `!$assignedExprType->isSuperTypeOf($propertyReflection->getWritableType())->yes()` instead. Please add a test and modify the implementation.

Also there might be a situation where the property is not defined at all. In that case we should simulate MethodCall to `__set` and see what throw points that produces (when the method exists). You can do that by calling `$this->processExprNode(new MethodCall($var->var, '__set'), $scope, static function (): void {}, $context)->getThrowPoints()`.

There's a similar bug that you can look to in the next PR :) https://github.com/phpstan/phpstan/issues/4852

Thank you!